Solutions/Corvus.Tenancy.Specs/Bindings/FakeTenantProvider.cs

```diff
@@ -15,17 +15,23 @@ public FakeTenantProvider(RootTenant rootTenant)
 
         public Task<ITenant> CreateChildTenantAsync(string parentTenantId)
         {
+#pragma warning disable RCS1079 // Throwing of new NotImplementedException.
```

Comment:
Since every method throws this, and given that the point of this class is apparently to provide an implementation that always throws, would a suppression attribute at class scope be a neater way to avoid the warnings?

Reply:
Hmmm. I don't like doing that sort of blanket disablement. This is definitely deliberately not implemented there. Rather than a new interface member I've auto-implemented and then forgot to *actually* implement.

Reply:
While in general I also dislike blanked disablement, in this particular case, I think this really is a case of not implemented being the default, and the exception being the things you have implemented. And what makes this unusual is that it's a fake object.

I'm actually wondering whether a strict mock would be better. `new Mock<ITenantProvider>(MockBehavior.Strict)` will throw if you invoke any methods that you've not defined an implementation for. (It actually throws a `MockException`, but are you actually depending on a `NotImplementedException` being thrown here? It's not clear whether you are.) That's arguably a more direct expression of what's going on here, i.e.: this is a fake that only implements one method, and if anything else gets invoked, then this test is all wrong.